torch/fx/graph.py

```diff
@@ -622,16 +622,33 @@ def process_outputs(self, out: Any) -> Any:
         return pytree.tree_unflatten(out, self.pytree_info.out_spec)
 
     def gen_fn_def(self, free_vars, maybe_return_annotation):
+        # Given a user function/model:
+        # ```python
+        # myargs = [myargs0, myargs1]
+        # mykwargs = {'mykwargs0': ..., 'mykwargs1': ...}
+        # def forward(self, mypos, *myargs, mykey=None, **mykwargs):
+        #
+        # the generated code flattens all keywords into positional arguments for `forward()`
+        # e.g forward(self, mypos, myargs0, myargs1, mykey, mykwargs0, mykwargs1):
+        # However, within `forward`, `tree_flatten_spec``still needs to parse positionals and keywords separately
+        # e.g. tree_flatten_spec(((mypos, myargs0, myargs1), {'mykey':mykey, 'mykwargs0':mykwargs0, 'mykwargs1':mykwargs1}), self._in_spec)
         if self.pytree_info is None:
             return super().gen_fn_def(free_vars, maybe_return_annotation)
+
         function_args = self.pytree_info.orig_args
         has_orig_self = (function_args[0] == 'self') if len(function_args) > 0 else False
         if has_orig_self:
             free_vars.insert(0, 'self')
         function_definition = super().gen_fn_def(function_args[:], maybe_return_annotation)
+
+        count_args = len(self.pytree_info.in_spec.children_specs[0].children_specs)
```

Comment:
`kwargs` handling is conflicting with `make_fx`, probably when `stateless.function_call` is used. 

This is an existing unit test (test_proxy_tensors.py) that fails with those changes

```python
    def test_make_fx_model_fwd_bwd(self):
        class Foo(torch.nn.Module):
            def __init__(self):
                super().__init__()
                self.linear = torch.nn.Linear(5, 5)

            def forward(self, x):
                return self.linear(x).relu()

        model = Foo()

        def f(x, params):
            out = stateless.functional_call(model, params, x).sum()
            out.backward()
            return list(params.values())
        input = torch.randn(3, 5, requires_grad=True)
        params = dict(model.named_parameters())
        fx_f = make_fx(f, tracing_mode=self.tracing_mode)(input, params)
```

Any idea on how to go about this? A naive idea would be to override `_PyTreeCodeGen` and its `gen_fn_def` method when `torch.export` is used, but maybe there is a more general solution